sdks/java/io/redis/src/main/java/org/apache/beam/sdk/io/redis/RedisConnection.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.redis;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.auto.value.AutoValue;
+
+import java.io.Serializable;
+import java.io.StringReader;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+
+import javax.annotation.Nullable;
+
+import org.apache.beam.sdk.transforms.display.DisplayData;
+
+import redis.clients.jedis.HostAndPort;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisCluster;
+import redis.clients.jedis.JedisCommands;
+
+/**
+ * {@code RedisConnection} describes and wraps a connection to Redis server or cluster.
+ */
+@AutoValue
+public abstract class RedisConnection implements Serializable {
+
+  @Nullable abstract List<String> nodes();
+  @Nullable abstract String password();
+  abstract int timeout();
+
+  abstract Builder builder();
+
+  @AutoValue.Builder
+  abstract static class Builder {
+    abstract Builder setNodes(List<String> nodes);
+    abstract Builder setPassword(String password);
+    abstract Builder setTimeout(int timeout);
+    abstract RedisConnection build();
+  }
+
+  public static RedisConnection create(List<String> nodes) {
+    checkArgument(nodes != null, "RedisConnection.create(nodes) called with null nodes list");
+    checkArgument(nodes.size() > 0, "RedisConnection.create(nodes) called with empty nodes list");
+    for (String node : nodes) {
+      checkArgument(node.contains(":"), "Invalid node syntax (not host:port): " + node);
+    }
+    return new AutoValue_RedisConnection.Builder().setNodes(nodes).setTimeout(3600).build();
+  }
+
+  public RedisConnection withPassword(String password) {
+    checkArgument(password != null, "RedisConnection.create(nodes).withPassword(password) "
+        + "called with null password");
+    return builder().setPassword(password).build();
+  }
+
+  public RedisConnection withTimeout(int timeout) {
+    return builder().setTimeout(timeout).build();
+  }
+
+  public Jedis connect() {
```

Comment:
Document this. Among other things, explain that it uses the first node in the list; by the way, is it allowed to call this method if there's more than 1 node? When should one call this method vs. connectCluster?